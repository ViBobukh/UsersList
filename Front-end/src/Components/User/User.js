import React, { useCallback, useState } from "react";
import List from "../List/List";
import classnames from "classnames";
import "./User.css";

/**
 * Component for render user
 * @param user {object}
 * @param isNest {boolean}
 * @returns {JSX.Element}
 */
function User({ user, isNest }) {
  const [isChildrenVisible, setChildrenVisible] = useState(false);

  /**
   * Callback for changing isChildrenVisible(opening and closing a nested user)
   * @type {function(*): void}
   */
  const onClick = useCallback(
    (event) => {
      event.stopPropagation();
      setChildrenVisible(!isChildrenVisible);
    },
    [isChildrenVisible]
  );

  return (
    <div className="user" onClick={onClick}>
      <div className={classnames({ nestUser: isNest })}>{user.name}</div>
      {isNest && isChildrenVisible ? <List users={user.items} /> : null}
    </div>
  );
}

export default User;
