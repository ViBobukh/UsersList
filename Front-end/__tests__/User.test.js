import React from "react";
import renderer from "react-test-renderer";
import User from "../src/Components/User/User";

describe("User tests", () => {
  it("Render user", () => {
    const user = {
      id: 1,
      name: "Maks",
      email: "maks@gmail.com",
    };
    const component = renderer.create(<User user={user} />);
    let domTree = component.toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
