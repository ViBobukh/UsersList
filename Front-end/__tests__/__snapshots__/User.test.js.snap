// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User tests Render user 1`] = `
<div
  className="user"
  onClick={[Function]}
>
  <div
    className=""
  >
    Maks
  </div>
</div>
`;
